src/expr/src/interpret.rs

```diff
@@ -151,6 +151,15 @@ impl<'a> ResultSpec<'a> {
         }
     }
 
+    /// Every result matches this spec.
+    pub fn anything_non_fallible() -> Self {
+        ResultSpec {
+            nullable: true,
+            fallible: false,
+            values: Values::All,
+        }
+    }
```

Comment:
I think this helper is basically cool, but I'd be tempted to (a) just call it `any_infallible` or something, and b) replace the existing `any_value.union(null)` call elsewhere in the file with a call to this.

Reply:
Done on the renaming, but I can't find this `any_value.union(null)` (or slight variations) in the file 🙈

Reply:
Here's one: https://github.com/MaterializeInc/materialize/blob/16bcd76cb9e803c88e9e9536f004fe405f5ac9a2/src/expr/src/interpret.rs#L579. That might be the only example though?

I think some of the other calls to `ResultSpec::anything` probably actually should be this narrower thing, but that would take some thinking and would be fine to punt!

Reply:
Now that I knew what to really look for I also didn't find other instances. Also, I took a look at other call sites of `ResultSpec::anything` and couldn't find and other low-hanging fruits (at least low-hanging for me) that I could replace.

Pushed an updated commit!